IBAnimatableApp/TransitionTableViewController.swift

```diff
@@ -44,47 +44,168 @@ class TransitionTableViewController: UITableViewController {
+    let portalAnimations: [TransitionAnimationType] = [.portal(direction: .forward, zoomScale: 0.3), .portal(direction: .backward, zoomScale: nil)]
+    transitionAnimations.append(toString(animations: portalAnimations))
+
     transitionAnimationsHeaders.append("NatGeo")
-    transitionAnimations.append(["NatGeo(Left)", "NatGeo(Right)"])
+    let natGeoAnimations: [TransitionAnimationType] = [.natGeo(to: .left), .natGeo(to: .right)]
+    transitionAnimations.append(toString(animations: natGeoAnimations))
+
     transitionAnimationsHeaders.append("Turn")
-    transitionAnimations.append(transitionTypeWithDirections(forName: "Turn"))
+    let turnAnimations: [TransitionAnimationType] = [.turn(from: .left), .turn(from: .right), .turn(from: .top), .turn(from: .bottom)]
+    transitionAnimations.append(toString(animations: turnAnimations))
+
     transitionAnimationsHeaders.append("Cards")
-    transitionAnimations.append(["Cards(Forward)", "Cards(Backward)"])
+    let cardAnimations: [TransitionAnimationType] = [.cards(direction: .forward), .cards(direction: .backward)]
+    transitionAnimations.append(toString(animations: cardAnimations))
+
     transitionAnimationsHeaders.append("Flip")
-    transitionAnimations.append(["Flip(Left)", "Flip(Right)"])
+    let flipAnimations: [TransitionAnimationType] = [.flip(from: .left), .flip(from: .right)]
+    transitionAnimations.append(toString(animations: flipAnimations))
+
     transitionAnimationsHeaders.append("Slide")
-    transitionAnimations.append(["Slide(Left, fade)", "Slide(Right)", "Slide(Top, fade)", "Slide(Bottom)"])
+    let slideAnimations: [TransitionAnimationType] = [.slide(to: .left, isFade: true), .slide(to: .right, isFade: false), .slide(to: .top, isFade: true), .slide(to: .bottom, isFade: false)]
+    transitionAnimations.append(toString(animations: slideAnimations))
+
     transitionAnimationsHeaders.append("Others")
-    transitionAnimations.append(["SystemRotate", "SystemRippleEffect", "SystemSuckEffect", "Explode", "Explode(10,-20,20)"])
+    let otherAnimations: [TransitionAnimationType] = [.systemRotate, .systemRippleEffect, .explode(xFactor: nil, minAngle: nil, maxAngle: nil), .explode(xFactor: 10, minAngle: -20, maxAngle: 20)]
+    transitionAnimations.append(toString(animations: otherAnimations))
   }
 
-  func transitionTypeWithDirections(forName prefixName: String) -> [String] {
-    return [prefixName + "(Left)", prefixName + "(Right)", prefixName + "(Top)", prefixName + "(Bottom)"]
+  //swiftlint:enable line_length
+
+  func toString(animations: [TransitionAnimationType]) -> [String] {
+    return animations.map { $0.asString }
   }
 
 }
 
+extension TransitionAnimationType {
+
+  var asString: String {
+    switch self {
+    case .fade(let direction):
+      return direction == .cross ? "Fade" : "Fade" + "(\(direction.asString))"
+    case .systemCube(let direction):
+      return "SystemCube" + "(\(direction.asString)"
+    case .systemFlip(let direction):
```

Comment:
I usually use a prefix `let` for more than one associated value. I can change it if you'd like :grin: